pkg/storage/engine/slice.s

```diff
@@ -1 +1,16 @@
-// Empty assembly file to allow go:linkname to work.
+// This empty assembly file has non-obvious side effects.
+//
+// 1. The go:linkname directive is only enabled for packages that
+// contain assembly files; this is the reason this file exists.
+//
+// 2. Completely empty assembly files cause GCC to mark the binary
+// as requiring an executable stack. This is a security risk. The
```

Comment:
Why?!?

Reply:
I'll rephrase this: it's nothing special about empty files, it's that any assembly file that doesn't have this marker is assumed to contain code that might rely on an executable stack. Empty files are a degenerate case that could be understood to not need executable stack, but that's not something that anyone really considered since there's not normally a reason for an empty file to exist.

Reply:
Ok, though I imagine empty assembly files are special in some regard since both the Go runtime and some third-party libraries we use such as `petermattis/goid` have non-empty assembly files which don't seem to cause a problem. My "Why?!?" comment was more about why GCC had such a subtle security behavior.

Reply:
The `.s` files used in `goid` are only used on older go versions. But there are other `.s` files that I'm pretty sure we do use that aren't having this effect. Mysterious.

Fun fact: `pkg/sql/exec` has [its own empty assembly file](https://github.com/cockroachdb/cockroach/blob/master/pkg/sql/exec/noescape.s), which evidently does not cause the executable to get the execstack bit. There is no `import "C"` in the package, so the `.s` file isn't even getting compiled. In current versions of go, it's not required to have a magic empty assembly file, it's required to [import unsafe](https://github.com/golang/go/blob/64c134f90f0cf6d0e55fca93c433b68810d12f12/src/cmd/compile/internal/gc/noder.go#L250) to use `go:linkname`. The "empty assembly file" pattern appears to be meaningless (I looked through blame as far as I could easily go and the logic about importing unsafe has been there since at least 2016). So I'm just going to delete both of these assembly files and keep the new test.